src/controllers/match2/index.ts

```diff
@@ -101,6 +94,76 @@ export class Match2Controller extends Controller {
 
     return responseWithAliases(match2)
   }
+
+  /**
+   * An optimiser creates the match2 {match2Id} on-chain. The match2 is now viewable to other members.
+   * @summary Create a new match2 on-chain
+   * @param match2Id The match2's identifier
+   */
+  @Post('{match2Id}/proposal')
+  @Response<NotFound>(404, 'Item not found')
+  @SuccessResponse('201')
+  public async createMatch2OnChain(@Path() match2Id: UUID): Promise<TransactionResponse> {
+    const [match2] = await this.db.getMatch2(match2Id)
+    if (!match2) throw new NotFound('match2')
+    if (match2.state !== Match2State.proposed) throw new BadRequest(`Match2 must have state: ${Match2State.proposed}`)
+
+    const [demandA] = await this.db.getDemand(match2.demandA)
+    validatePreOnChain(demandA, DemandSubtype.order, 'DemandA')
+
+    const [demandB] = await this.db.getDemand(match2.demandB)
+    validatePreOnChain(demandB, DemandSubtype.capacity, 'DemandB')
```

Comment:
since you are creating a helper method for each new db query, who not something like

```js
// pseudo code
const [demandA, demandB] = await this.db.getDemandAB(match2.demandA, match2.demandB)
// and validate
validatePreOnChain(await this.db.getDemandAB(match2.demandA, match2.demandB)
// or ...
await this.db.getDemandAB(match2.demandA, match2.demandB).foEach(demand => validatePreOnChain(....)
...
```

Reply:
what I've done is more readable to me, even if it repeats a bit. And I don't need to make a new `DemandAB`  function

Reply:
ok